### DISCUSSION TITLE
Validate MDX syntax before PR merge

### DISCUSSION THREAD

**Discussion by petrnetusil** [mdx-js/mdx#2287] [Q&A] [answered]
**Context:**
I'm running a documentation site, which is contributed to by many people. Contributors use MDX syntax and then raise a Pull Requests when they are ready. As part of a validation of such Pull Requests, I would love to have a validation that a correct MDX syntax is applied.

**Challenge:**
People sometimes contribute with a syntax that will result in a crashed application build. E.g. improperly written `<` or `>`. I would like to give them a meaningful message within their Pull Request (let's assume on GitHub), so that they can amend their code before we merge the PR.

**Question:**
What would be a best way to achieve this?

**Solutions conceived:**
As part of my CI/CD pipeline, always build the entire website. If the build fails, create a way how people can access the console output and then manually go through all the syntax issues.

**Solutions I'm hoping for 😄 :**
Script / parser, which would validate MDX syntax (e.g. on my CI/CD server) and return back an understandable and actionable response on what to fix. E.g. as part of new PR comment.

Thank you for any ideas!

**Comment by ChristianMurphy** [comment, marked as answer]
> Script / parser, which would validate MDX syntax (e.g. on my CI/CD server) and return back an understandable and actionable response

The MDX parser includes positional information which can be used to determine where the issue happened.
You can see this as an example of this on the MDX site.
![Screenshot 2023-04-21 at 12-33-59 Playground MDX](https://user-images.githubusercontent.com/3107513/233720133-05723204-10aa-4d7e-943a-e352b9e28b29.png)

> understandable and actionable response on what to fix.

Fixable sounds like it's getting into the realm of https://github.com/mdx-js/mdx-analyzer
Which can run as a language server in an editing experience (implementation for VSCode, and is ready to try) or _could_ be run as a CLI (not implemented yet, see https://github.com/mdx-js/mdx-analyzer/issues/292)

**Reply by petrnetusil** [reply]
Thanks! Before the CLI is done, we’ve ended up with some simple MD/HTML/JS parser first to hopefully catch as much as possible already.
